### PR TITLE
Support time in index key

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

**Review by dutor — APPROVED** [review]
LGTM.

File: src/utils/IndexKeyUtils.h [hunk 1]
```diff
@@ -183,15 +191,43 @@ class IndexKeyUtils final {
         return val;
     }
 
+    static std::string encodeTime(const nebula::Time& t) {
```
Comment (Shylock-Hg) [inline src/utils/IndexKeyUtils.h:194]: Pls add a TODO about convert to UTC.

File: src/utils/IndexKeyUtils.h [hunk 2]
```diff
@@ -183,15 +191,43 @@ class IndexKeyUtils final {
...
+        std::string buf;
+        buf.reserve(sizeof(int32_t) + sizeof(int8_t) * 3);
+        buf.append(reinterpret_cast<const char*>(&hour), sizeof(int8_t))
+           .append(reinterpret_cast<const char*>(&minute), sizeof(int8_t))
+           .append(reinterpret_cast<const char*>(&sec), sizeof(int8_t))
+           .append(reinterpret_cast<const char*>(&microsec), sizeof(int32_t));
+        return buf;
+    }
+
+    static nebula::Time decodeTime(const folly::StringPiece& raw) {
```
Comment (Shylock-Hg) [inline src/utils/IndexKeyUtils.h:208]: Pls add a TODO about convert to current timezone.

File: src/utils/IndexKeyUtils.h [hunk 3]
```diff
@@ -183,15 +191,43 @@ class IndexKeyUtils final {
...
+
+        nebula::Time t;
+        t.hour = folly::Endian::big8(hour);
+        t.minute = folly::Endian::big8(minute);
+        t.sec = folly::Endian::big8(sec);
+        t.microsec = folly::Endian::big32(microsec);
+        return t;
+    }
+
     static std::string encodeDate(const nebula::Date& d) {
```
Comment (Shylock-Hg) [inline src/utils/IndexKeyUtils.h:222]: ditto.

**Review by laura-ding — APPROVED** [review]

File: src/utils/IndexKeyUtils.h [hunk 4]
```diff
@@ -216,12 +252,12 @@ class IndexKeyUtils final {
         std::string buf;
```
Comment (Shylock-Hg) [inline src/utils/IndexKeyUtils.h:252]: ditto.

**Pull request merged**